base/linalg/matmul.jl

```diff
@@ -44,7 +44,7 @@ end
 vecdot{T<:BlasReal}(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) = BLAS.dot(x, y)
 vecdot{T<:BlasComplex}(x::Union{DenseArray{T},StridedVector{T}}, y::Union{DenseArray{T},StridedVector{T}}) = BLAS.dotc(x, y)
 
-function dot{T<:BlasReal, TI<:Integer}(x::Vector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::Vector{T}, ry::Union{UnitRange{TI},Range{TI}})
+function dot{T<:BlasReal, TI<:Integer}(x::StridedVector{T}, rx::Union{UnitRange{TI},Range{TI}}, y::StridedVector{T}, ry::Union{UnitRange{TI},Range{TI}})
```

Comment:
No `DenseArray` is not a subtype of `StridedVector. The reason we have this is to be able to handle e.g. a matrix as a vector without explicitly reshaping it. I think it would be cleaner if we could simply reshape without cost. I'm now sure how often people use this feature for `dot` but it was requested for `axpy` at some point.